modules/packages/LinearAlgebra.chpl

```diff
@@ -1572,6 +1572,17 @@ module Sparse {
     compilerError("Matrix division not yet supported for sparse matrices */");
   }
 
+  /* Return an identity matrix over sparse domain ``Dom`` */
+  proc eye(Dom, type eltType=real) where isCSDom(Dom) {
+    const (m,n) = Dom.shape;
+    var D = CSRDomain(m,n);
```

Comment:
I know you're following the dense `eye()` implementation, but I just noticed an issue - this does not preserve the input `Dom` indices. Perhaps this line should do `var D = CSRDomain(Dom.parentDom)` instead.

Reply:
Quick question -- what does `Dom.parentDom` return?

Reply:
I think I answered my last question, but here's another -- are we requiring that matrix indices start at 1?

Reply:
> I think I answered my last question, but here's another -- are we requiring that matrix indices start at 1?

The `LinearAlgebra` library's stance is: We'll preserve indices where we can, otherwise we'll give you 1-based indices. Implementations should not be index-dependent.

Reply:
and if that stance is not fulfilled anywhere, I would consider it a bug.

Reply:
Ok, well -- I think `_eyeDiagonal` isn't strictly correct then for 
```
proc eye(Dom: domain(2), type eltType=real) {
```
overload of `eye`, right?

Reply:
Ignore the last comment -- I was being an idiot; the code is fine.

Reply:
I almost agreed with you :) That's a tricky one.